### PR TITLE
[AMORO-1449]: Extract common spark module  to reuse code between different spark versions

### LINKED ISSUE #1449: [Improvement]: Extract Spark Common Module to reuse code between different spark versions.
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Too much duplicate codes in spark 3.x modules. we can extract a common module to reuse those codes.

### How should we improve?

Extract a spark common module to reuse codes:

1. to reuse tests like SQL tests.
2. to reuse util class which dosen't matter with spark version or depend on stable api

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

fix #1449 

## Brief change log

- Add a common spark module.
- Extract common test framework to common spark module
- Extract properties to reuse between spark versions.

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] Run test locally before making a pull request

### CODE REVIEW

File: core/src/test/java/com/netease/arctic/CatalogMetaTestUtil.java [hunk 1]
```diff
@@ -62,6 +62,37 @@ public static CatalogMeta createArcticCatalog(File arcticBaseDir) throws IOExcep
     return meta;
   }
 
+  public static CatalogMeta createHadoopCatalog(File arcticBaseDir) throws IOException {
```
Comment (zhoujinsong) [inline core/src/test/java/com/netease/arctic/CatalogMetaTestUtil.java:65]: As the module core use `CatalogTestHelpers#buildCatalogMeta` to build the catalog meta, would you like to move this class to the spark common module?

Or we should try to unity the catalog meta construct util across modules.

File: spark/common/src/test/java/com/netease/arctic/spark/test/SparkTestBase.java [hunk 2]
```diff
@@ -60,7 +59,7 @@ public static void tearDownContext() {
 
   protected Map<String, String> sparkSessionConfig() {
     return ImmutableMap.of(
-        "spark.sql.catalog.spark_catalog", ArcticSparkSessionCatalog.class.getName(),
+        "spark.sql.catalog.spark_catalog", "com.netease.arctic.spark.ArcticSparkSessionCatalog",
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply spark/common/src/test/java/com/netease/arctic/spark/test/SparkTestBase.java]: done

File: core/src/test/java/com/netease/arctic/CatalogMetaTestUtil.java (hunk 1 above)
Comment (baiyangtx) [reply core/src/test/java/com/netease/arctic/CatalogMetaTestUtil.java]: This method and class are removed, using CatalogTestHelper instead.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**